go/vt/vttablet/tabletserver/connpool/dbconn_test.go

```diff
@@ -212,15 +195,11 @@ func TestDBConnDeadline(t *testing.T) {
 	if dbConn != nil {
 		defer dbConn.Close()
 	}
-	if err != nil {
-		t.Fatalf("should not get an error, err: %v", err)
-	}
+	require.NoError(t, err)
 
 	_, err = dbConn.Exec(ctx, sql, 1, false)
-	want := "context deadline exceeded before execution started"
-	if err == nil || !strings.Contains(err.Error(), want) {
-		t.Errorf("Exec: %v, want %s", err, want)
-	}
+	require.Error(t, err)
+	require.ErrorContains(t, err, "(errno 3024) (sqlstate HY000): Query execution was interrupted, maximum statement execution time exceeded, before execution started")
 
```

Comment:
Let us change the last line to `before executing on underlying database`.

Reply:
I think the current message is fine. Why "underlying database"?? I'd just drop the comma between exceeded and before.

Reply:
Left this as suggestions on the appropriate LOC.

Reply:
I see that execution started at the VTGate and VTTablet, but not on MySQL, so adding that message would make it clear.